### PR TITLE
[2.9] Add OVS integration test

### PR DESCRIPTION
**Pull request by achilleasa**
This PR adds a regression test for ensuring that Juju correctly retains empty `openvswitch` blocks (such as the ones injected by MAAS to composed KVM machines with OVS bridges) when merging netplan configurations.

## QA steps

The test is wired to use a dual-spaced machine with an OVS bridge on our internal finfolk MAAS.
To run the QA steps, define a MAAS cloud and point it to finfolk. Then:

```sh
$ cd tests
$ ./main.sh -v -p maas -c $finfolk-cloud-name ovs_maas
```

### Setting up a suitable machine for this test on your local MAAS instance

The following steps will allow you to prepare a machine with the right NIC devices for running the above test. The network setup is based on [this](https://openstackdevops.wordpress.com/2021/05/03/converged-networking-with-charmed-openstack/) blog post about converged networking.

Prior to running the following steps you must:
 - be running MAAS 2.9.2+.
 - define two spaces named **space1** and **space2**.

Using the MAAS UI:
1) Compose a new KVM machine (1 core, 4G RAM)
2) **Abort** the commissioning attempt for the new machine.
3) Connect to the KVM host and edit the machine definition using `virsh`. For example, if your machine is called 'foo' run:
    `virsh edit foo`.
4) Locate the XML block that describes the one and only NIC assigned to the machine. It should look something like:
   ```xml
     <interface type='network'>
      <mac address='52:54:00:24:a7:07'/>
      <source network='maas'/>
      <model type='virtio'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x04' function='0x0'/>
    </interface>
   ```
5) Clone this block two times (and change the **slot** attribute to **0x08** and **0x0a** respectively). Use [this](https://gist.github.com/viz3/6591201) script to generate a MAC for each new NIC.
6) Go back to the MAAS UI and commission the machine.
7) Edit the machine and attach the `ovs` tag to it.
8) In the networking tab for the machine you should now see three NICs.
    - Select the first two NICS and click the `create bond` button; leave the default values as-is and save the bond.
    - Select the bond and click the `create bridge` button. Under bridge type select `open vswitch`. Then, select the appropriate fabric / subnet values for `space1` and `Auto assign` for the IP configuration option.
    - Select the remaining NIC from the list, edit it and select the appropriate fabric/subnet values for `space2` and `Auto assign` for the IP configuration option.
9) Deploy the machine and ensure that the operation succeeds. If the machine gets stuck in `Rebooting` ask the MAAS folks for help.

### CODE REVIEW

File: tests/suites/ovs_maas/task.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+	set_verbosity
+
+	echo "==> Checking for dependencies"
+	check_dependencies juju
+
+	file="${TEST_DIR}/test-ovs-maas.log"
+
+	# shellcheck disable=SC2140
+  export BOOTSTRAP_ADDITIONAL_ARGS="${BOOTSTRAP_ADDITIONAL_ARGS:-} --bootstrap-series=focal --bootstrap-constraints="tags=ovs""
```
Comment (SimonRichardson) [inline tests/suites/ovs_maas/task.sh:15]: Whitespace issue.

**Review by SimonRichardson — COMMENTED** [review]
can you run shfmt on this.

**Comment by manadart** [comment]
I am getting this when I run main.
```
Only clouds with registered credentials are shown.
There are more clouds, use --all to see them.
Only clouds with registered credentials are shown.
There are more clouds, use --all to see them.
Only clouds with registered credentials are shown.
There are more clouds, use --all to see them.
jq: error (at <stdin>:1): null (null) has no keys
```

**Comment by achilleasa** [comment]
> I am getting this when I run main.
> 
> ```
> Only clouds with registered credentials are shown.
> There are more clouds, use --all to see them.
> Only clouds with registered credentials are shown.
> There are more clouds, use --all to see them.
> Only clouds with registered credentials are shown.
> There are more clouds, use --all to see them.
> jq: error (at <stdin>:1): null (null) has no keys
> ```

Looks like something that the test runner should be suppressing? I don't believe it is related to the changes from this PR.

**Review by manadart — APPROVED** [review]
Worked once I added a region to my cloud. 

I think we could probably revert the last commit suppressing the show-cloud stderr given that wasn't the issue, but we also probably don't need that fluff in test output...

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**